### PR TITLE
Add empty check for subscription name

### PR DESCRIPTION
**Pull request by nkurihar**
### Motivation
Subscription name can be empty for now.
It causes some problems.

(Indeed, recently we encountered sudden Broker stopping caused by large number of requests 　whose subscription name is empty through WebSocketProxy)

### Modifications
* Add empty check for subscription name to `PersistentTopic`, `PulsarClientImpl` and `ConsumerHandler`
* Add test cases for empty subscription name

### Result
* Subscription name can not be empty.

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java [hunk 1]
```diff
@@ -349,6 +350,12 @@ public void removeProducer(Producer producer) {
 
         final CompletableFuture<Consumer> future = new CompletableFuture<>();
 
+        if (isBlank(subscriptionName)) {
+            log.warn("[{}] Empty subscription name", topic);
```
Comment (merlimat) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java]: Could we leave this at debug level in the broker logs?

Comment (nkurihar) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java]: @merlimat I changed the log level to debug.
